tools/metactl/src/main.rs

```diff
@@ -169,3 +184,12 @@ fn print_meta() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+// if port is open, service is running
+async fn service_is_running(addr: &str) -> anyhow::Result<bool> {
```

Comment:
`anyhow::Result` will spread everywhere. Only use it at the top-level entry.

Here a specific error type would be better.

Reply:
Updated